websocket-chat-broker/src/main.rs

```diff
@@ -0,0 +1,156 @@
+        let content = format!("{}: {}",
+                              self.name.clone().unwrap_or("anon".to_string()),
+                              msg);
+        let msg = SendMessage(self.room.clone(), self.id, content);
+        self.issue_async(msg);
+    }
+}
+
+impl Actor for WsChatSession {
+    type Context = ws::WebsocketContext<Self>;
+
+    fn started(&mut self, ctx: &mut Self::Context) {
+        self.join_room("Main", ctx);
+    }
+
+    fn stopped(&mut self, _ctx: &mut Self::Context) {
+        println!("WsChatSession stopped for {}({}) in room {}",
+            self.name.clone().unwrap_or("anon".to_string()),
+            self.id,
+            self.room);
+    }
+}
+
+impl Handler<ChatMessage> for WsChatSession {
+    type Result = (); 
+
+    fn handle(&mut self, msg: ChatMessage, ctx: &mut Self::Context) {
+        ctx.text(msg.0);
+    }
+}
+
+impl StreamHandler<ws::Message, ws::ProtocolError> for WsChatSession {
+    fn handle(&mut self, msg: ws::Message, ctx: &mut Self::Context) {
+        println!("WEBSOCKET MESSAGE: {:?}", msg);
+        match msg {
+            ws::Message::Text(text) => {
+                let m = text.trim();
+                if m.starts_with('/') {
+                    let v: Vec<&str> = m.splitn(2, ' ').collect();
+                    match v[0] {
+                        "/list" => self.list_rooms(ctx),
+                        "/join" => {
+                            if v.len() == 2 {
+                                self.join_room(v[1], ctx);
+                                ctx.text("joined");
+                            } else {
+                                ctx.text("!!! room name is required");
+                            }
+                        }
+                        "/name" => {
+                            if v.len() == 2 {
+                                self.name = Some(v[1].to_owned());
+                                ctx.text(format!("name changed to: {}", v[1]));
+                            } else {
+                                ctx.text("!!! name is required");
+                            }
+                        }
+                        _ => ctx.text(format!("!!! unknown command: {:?}", m)),
+                    }
+                    return;
+                } 
+                self.send_msg(m);
+            }
+            ws::Message::Close(_) => {
+                ctx.stop();
+            },
+            _ => {},
+        }
+    }
+}
+
+fn main() {
+    let sys = actix::System::new("websocket-broker-example");
+
+    HttpServer::new(move || {
+        App::new()
+            .resource("/", |r| r.method(http::Method::GET).f(|_| {
+                HttpResponse::Found()
+                    .header("LOCATION", "/static/websocket.html")
+                    .finish()
+            }))
+            .resource("/ws/", |r| r.route().f(chat_route))
+            .handler("/static/", fs::StaticFiles::new("static/").unwrap())
+    }).bind("127.0.0.1:8080")
```

Comment:
Just to be sure, could you please run rustfmt?

Reply:
Almost forgot but done!